### DISCUSSION TITLE
Entities of single device cannot be controlled independently using scene editor

### DISCUSSION THREAD

**Discussion by kongo09** [home-assistant/frontend#10105] [Configuration] [answered]
### The problem

The scene editor does not allow individual entities to be added to a scene when they belong to a device that provides more than one entity. The editor always adds all entities of the device, even if the user tries to add a single entity in the `Add Entity` dialog.

In the example below I only tried to put the first entity into the scene, not the second:

![image](https://user-images.githubusercontent.com/640121/135219450-52c84bfa-dd7b-49a4-a1a5-1ce8b19d4b4f.png)

As a result, certain scene configurations are impossible to control from the UI. It seems to be possible, however, to add individual entities from the `scenes.yaml` file by hand. That's why I consider this a bug.

### What is version of Home Assistant Core has the issue?

core-2021.9.7

### What was the last working version of Home Assistant Core?

never worked

### What type of installation are you running?

Home Assistant Supervised

### Integration causing the issue

scene

### Link to integration documentation on our website

https://www.home-assistant.io/integrations/scene/

**Comment by probot-home-assistant** [comment]
Hey there @home-assistant/core, mind taking a look at this issue as it has been labeled with an integration (`scene`) you are listed as a [code owner](https://github.com/home-assistant/core/blob/dev/CODEOWNERS#L444) for? Thanks!
<sub><sup>(message by CodeOwnersMention)</sup></sub>

---

[scene documentation](https://www.home-assistant.io/integrations/scene)
[scene source](https://github.com/home-assistant/core/tree/dev/homeassistant/components/scene)
<sub><sup>(message by IssueLinks)</sup></sub>

**Comment by emontnemery** [comment]
This is a limitation in the frontend's scene editor, the resulting scene has no concept of devices.
For devices with multiple switch or ligth entities, like in the screenshot, it makes sense to not force-include all of them.

Maybe frontend can show all entities of a device, but allow excluding an entity from the actual scene?

**Comment by kongo09** [comment]
Suggestion for a user-friendly solution consistent with the current interface would be:

- using `Add Device` adds the whole device like it is doing already, including all entities
- using `Add Entity` only adds the selected entity

Adding a device when the user explicitely asked to add an entity feels like a bug from the user perspective.

**Comment by kongo09** [comment]
Possibly related, though I consider this more a bug than a feature request:
https://github.com/home-assistant/frontend/issues/6468

**Comment by kongo09** [comment]
Looking at the code, I understand this behaviour has been introduced by purpose in PR https://github.com/home-assistant/frontend/pull/6343 by @bramkragten so there is probably some higher intention to this?

**Comment by emontnemery** [comment]
The intention is to make the scene editor device oriented, but it doesn't mean we should force a scene to actuate all devices.

Maybe something like this would work:
![image](https://user-images.githubusercontent.com/14281572/135246254-e31cf328-b699-44ee-9955-227f1d7c334b.png)

**Comment by kongo09** [comment]
Understood. However, while the scene editor is _not really_ device oriented just now, would a roll-back of the above mentioned code not solve the problem, or are there other things to consider? It could be re-introduced once the UI reflects the desired device orientation.

**Comment by emontnemery** [comment]
I think the UI is OK as is, but we should allow excluding entities of a device from a scene.

**Comment by DieterKoblenz** [comment]
Also see https://community.home-assistant.io/t/cant-add-single-entity-to-scene/343505

I support changing it back to excluding entities in the UI (workaround is via YAML).

**Comment by Ragingfire105** [comment]
In the current state of the scenes ui, the scenes editor becomes a landmine waiting to be stepped on if you have used the yaml work around.  Scenes to me seem to be a core function for a smart home platform.  

I wish this issue had more attention paid to it considering the obvious push to more UI based configuration to improve user adoption through easy of use.

My problem devices are Shelly 2.5s with a light and fan.  Making a lighting scene shouldn't force me to also change the fan.

**Comment by emontnemery** [comment]
There's a draft PR to fix this: https://github.com/home-assistant/frontend/pull/11046

**Comment by davet2001** [comment]
I ended up here with a use case that makes this even more annoying.

RPi RF was dropped as of 2022.4.  I was using this to control lots of 433MHz RF mains switches.
As suggested, I created an ESPHome device with RF switches configured - tested all working - ready to replace RPi RF.

But when I come to put some of those switches into scenes... as per above, it auto includes the esphome device which includes _all_ of my RF switches.  So now any scene must set the state of all RF switches in my house.  Which is never what I want to do.

**Comment by benjycov** [comment]
I also have a big problem with scenes automatically adding the whole device. I want to use scenes for my lights, some of which are on ESPs which control other things.

In the example below I have an ESP which does lots of quite important spa control related things - which I definitely do not want my scene touching, the same ESP happens to also to control the pool and spa lights - which I want in my scene.

![image](https://user-images.githubusercontent.com/70473832/170989384-c2ca9406-04bc-4e13-950b-f82e76f14edd.png)

**Comment by emontnemery** [comment, marked as answer]
This has been fixed by https://github.com/home-assistant/frontend/pull/11046, which is included in Home Assistant Core 2022.06

**Comment by marionspd** [comment]
My home assistant version is updated to 2022.06 and I don't get the option to select entities, only devices.  Any suggestions as to what I might be doing wrong?

edit: added proper image
![MyHANewScene](https://user-images.githubusercontent.com/107815573/175203277-75781c76-ba7a-4a39-9abd-1c4d53c46f7b.png)

**Reply by frenck** [reply]
It is an advanced feature, which will only show up when you have enabled advanced mode in your user profile.
